### PR TITLE
Update JDK Version (#771)

### LINKED ISSUE #771: Correct detail on JDK support
Currently Besu does not support JDK 17, so the statement that JDK 11+ are supported is not true. This may require further investigation. See hyperledger/besu#2617

### PR DESCRIPTION
**Pull request by rolandtyler**
Signed-off-by: Roland Tyler <roland.tyler@consensys.net>

## Pull request checklist

Use the following list to make sure your PR fits the Besu doc quality standard.

### Before creating the pull request

Make sure that:

- [ ] [all commits in this PR are signed off for the DCO](https://wiki.hyperledger.org/display/BESU/DCO).
- [ ] you read the [contribution guidelines](https://wiki.hyperledger.org/display/BESU/Contributing+to+documentation).
- [ ] you have [tested your changes locally](https://wiki.hyperledger.org/display/BESU/MkDocs+And+Markdown+Guide#MkDocsAndMarkdownGuide-PreviewTheDocumentation) before submitting them to the community for review.

### After creating your pull request and tests finished

Make sure that:

- [ ] you fixed all the issues raised by the tests, if any.
- [ ] you verified the rendering of your changes on [ReadTheDocs.org PR preview](https://wiki.hyperledger.org/display/BESU/MkDocs+And+Markdown+Guide#MkDocsAndMarkdownGuide-PreviewwithReadTheDocs)
  and updated the testing link (see [Testing](#testing)).

## Describe the change
Update JDK version references. Remove outdated error message from previous version.

## Issue fixed
Fixes #771 

### For content changes

- [ ] Doc content
- [ ] Doc pages organisation

### For tools changes

- [ ] CircleCI workflow
- [ ] Build and QA tools (lint, vale,…)
- [ ] MkDocs templates
- [ ] MkDocs configuration
- [ ] Python dependencies
- [ ] Node dependencies and JavaScript
- [ ] ReadTheDocs configuration
- [ ] GitHub integration

### CODE REVIEW

File: docs/HowTo/Get-Started/Installation-Options/Install-Binaries.md [hunk 1]
```diff
@@ -16,8 +16,8 @@ description: Install or upgrade Hyperledger Besu from binary distribution
     Hyperledger Besu requires:
 
       * MacOS High Sierra 10.13 or later versions
-      * Java 11+ to compile. Earlier versions are not supported. You can install Java using
-        `brew install openjdk`. Alternatively, you can manually install the
+      * Java versions 11 through 16 to compile. Earlier versions and version 17 are not supported.
```
Comment (alexandratran) [inline docs/HowTo/Get-Started/Installation-Options/Install-Binaries.md:19]:
```suggestion
      * Java version 11-16 to compile. Earlier versions and version 17 are not supported.
```

File: docs/HowTo/Get-Started/Installation-Options/Install-Binaries.md [hunk 2]
```diff
@@ -65,7 +65,7 @@ besu --help
 
 !!!attention
 
-    Besu requires Java 11+ to compile. Earlier versions are not supported.
+    Besu requires Java 11 thorugh 16 to compile. Earlier versions and version 17 are not supported.
```
Comment (alexandratran) [inline docs/HowTo/Get-Started/Installation-Options/Install-Binaries.md:68]:
```suggestion
    Besu requires Java version 11-16 to compile. Earlier versions and version 17 are not supported.
```

**Review by alexandratran — COMMENTED** [review]
Suggestion.

**Review by alexandratran — APPROVED** [review]

File: docs/HowTo/Get-Started/Installation-Options/Install-Binaries.md [hunk 3]
```diff
@@ -13,10 +13,10 @@ description: Install or upgrade Hyperledger Besu from binary distribution
 
 !!!important
 
-    Hyperledger Besu requires:
+    Hyperledger Besu supports:
 
       * MacOS High Sierra 10.13 or later versions
```
Comment (alexandratran) [inline docs/HowTo/Get-Started/Installation-Options/Install-Binaries.md:18]:
```suggestion
      * MacOS High Sierra 10.13 or later versions.
```

File: docs/HowTo/Get-Started/Installation-Options/Install-Binaries.md [hunk 4]
```diff
@@ -13,10 +13,10 @@ description: Install or upgrade Hyperledger Besu from binary distribution
...
-      * Java version 11-16 to compile. Earlier versions and version 17 are not supported.
+      * Java version 11-16.
```
Comment (alexandratran) [inline docs/HowTo/Get-Started/Installation-Options/Install-Binaries.md:19]:
```suggestion
      * Java 11-16.
```

File: docs/HowTo/Get-Started/Installation-Options/Install-Binaries.md [hunk 5]
```diff
@@ -65,7 +65,7 @@ besu --help
 
 !!!attention
 
-    Besu requires Java version 11-16 to compile. Earlier versions and version 17 are not supported.
+    Besu supports Java version 11-16.
```
Comment (alexandratran) [inline docs/HowTo/Get-Started/Installation-Options/Install-Binaries.md:68]:
```suggestion
    Besu supports Java 11-16.
```

**Review by ajsutton — APPROVED** [review]
LGTM. We still aren't separating the compile time requirements from the runtime requirements - at the moment they're the same but when JDK 17 is released we'll likely be able to run on it but not compile on it.

**Pull request merged**